cmd/url.go

```diff
@@ -87,8 +89,24 @@ The created URL can be used to access the specified component from outside the O
 			os.Exit(1)
 		}
 
+		componentPorts, err := url.GetComponentServicePortNumbers(client, componentName, applicationName)
+		checkError(err, "unable to get component exposed ports")
+
+		var portFound bool
+
+		for _, port := range componentPorts {
+			if urlPort == port {
```

Comment:
Is flag corresponding to urlPort mandatory? It seems so from line 103 below(https://github.com/redhat-developer/odo/pull/615/files#diff-626187d5d8d29a6936cc1b5876f7edd9R103)
As I can see from https://github.com/redhat-developer/odo/issues/553 , this is required for cases where a component has multiple ports exposed and to be to correctly choose which port to expose...
In which case, should we have a default no flag required case when there's only 1 port the component has and also accordingly either error out if flag not passed even though there are multiple ports or retain the current round robin choice of port if flag is not passed?
Please correct me if I am wrong as I might be missing something

Reply:
Yes I have made the `--port` flag to avoid the round robin problem stated in the issue. But incase of 1 port service component, I think we can retrieve the port exposed and make it the default if no port is passed. @kadel Should I make the port flag completely mandatory?

Reply:
> Yes I have made the --port flag to avoid the round robin problem stated in the issue.

Yes

> But incase of 1 port service component, I think we can retrieve the port exposed and make it the default if no port is passed

Exactly what I meant, but it doesn't appear to be the case looking at: 
https://github.com/redhat-developer/odo/pull/615/files/b2b39f5a74142e815d8d25b91823faf48a2d93c3#diff-626187d5d8d29a6936cc1b5876f7edd9R217 it appears, that the default port value if not passed is -1(I understand the reason for choice is AFAIK port numbers are positive and hence to be able to differentiate a valid port number from a non-passed port number) which is not checked for to assume a default port(can be the only obvious port in case of single port or a random selection of port in case of multiple ports) as in, if you see line 98, it only validates that the port provided by user as a cli flag, is a valid port from list of component ports but I don't see a place where port not being passed by user is handled

Reply:
The current PR does not implement the default single port selection as of now. Currently the flag is marked as required. @kadel Should I implement the single port automatic selection or leave the flag required as marked?

Reply:
> . @kadel Should I make the port flag completely mandatory?

no flag shouldn't be mandatory if a component is exposing just one port. In that case odo should detect correct port automatically.